src/backend/gl/src/lib.rs

```diff
@@ -37,15 +39,91 @@ mod state;
 mod window;
 
 #[cfg(feature = "glutin")]
-pub use window::glutin::{config_context, Headless, Surface, Swapchain};
+pub use window::glutin::{Surface, Swapchain, Window};
+
+#[cfg(feature = "glutin")]
+pub struct Instance {
+    el: Arc<Mutex<glutin::EventsLoop>>,
+    context: glutin::Context,
+    pub(crate) quadvbo: Arc<Mutex<Option<u32>>>,
+    pub(crate) quadprogram: Arc<Mutex<Option<u32>>>,
+}
+
+#[cfg(feature = "glutin")]
+pub enum InstanceOrWindow {
+    Instance(Arc<Instance>),
+    Window(Starc<glutin::GlWindow>),
+}
+
+#[cfg(feature = "glutin")]
+impl Deref for InstanceOrWindow {
+    type Target = glutin::Context;
+    fn deref(&self) -> &Self::Target {
+        match self {
+            InstanceOrWindow::Instance(i) => &i.context,
+            InstanceOrWindow::Window(w) => w.context(),
+        }
+    }
+}
+
+#[cfg(feature = "glutin")]
+impl Instance {
+    pub fn create(_name: &str, _version: u32, el: Arc<Mutex<glutin::EventsLoop>>) -> Arc<Self> {
+        let cb = glutin::ContextBuilder::new();
+        let context = glutin::Context::new(cb, &el.lock().unwrap(), true).unwrap();
+
+        Arc::new(Instance {
+            el,
+            context,
+            quadvbo: Arc::new(Mutex::new(None)),
+            quadprogram: Arc::new(Mutex::new(None)),
+        })
+    }
+
+    pub fn create_surface(&self, window: &Arc<Mutex<Window>>) -> Arc<Surface> {
+        Arc::new(Surface {
+            window: Arc::clone(window),
+        })
+    }
+
+    pub fn create_window(&self, wb: glutin::WindowBuilder) -> Arc<Mutex<Window>> {
+        Window::new(wb)
+    }
+}
+
+#[cfg(feature = "glutin")]
+impl Drop for Instance {
+    fn drop(&mut self) {
+        unsafe {
+            let gl = gl::Gl::load_with(|s| self.context.get_proc_address(s) as *const _);
```

Comment:
why are we loading GL pointers here?

Reply:
Because the instance doesn't have a copy of the share or the gl ptrs, so it's easier (and still correct) to just load them in again.